### PR TITLE
Add support for Oauth to enable GCS access

### PR DESCRIPTION
**Pull request by dcfocus**
Refer to [old PR](https://github.com/prestodb/presto-python-client/pull/97) for comments.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,8 @@ python:
   - "2.7"
   - "3.5"
   - "3.6"
-  - "pypy-5.4"
+  - "3.7"
+  - "pypy3.6-7.0.0"
```
Comment (ChunxuTang) [inline .travis.yml:7]: The previous python versions seem to be a little outdated. I guess that's why you'd like to update the versions. 
May we move the change to a separate PR? This PR concentrates on the OAuth-related mechanism instead of python versions.

File: prestodb/client.py [hunk 2]
```diff
@@ -422,6 +434,15 @@ def process(self, http_response):
...
         )
 
+    @property
+    def http_session(self):
+        return self._http_session
+
+    def get_oauth_token(self):
+        self.credentials.refresh(self.auth_req)
+        return {
+            constants.PRESTO_EXTRA_CREDENTIAL: "%s = %s" % (constants.GCS_CREDENTIALS_OAUTH_TOKEN_KEY, self.credentials.token)
```
Comment (ChunxuTang) [inline prestodb/client.py:444]: This line seems to be too long. I noticed that the tool `black` is used at this codebase for formatting. Would you mind formatting your code with `black`?
A reference: https://github.com/prestodb/presto-python-client/commit/d48f458799982909111c76707127835c3836f928.

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,py35,py36,pypy2
+envlist = py27,py35,py36,py37,pypy2
```
Comment (ChunxuTang) [inline tox.ini:2]: Another python version update here.

File: .travis.yml (hunk 1 above)
Comment (dcfocus) [reply .travis.yml:7]: The old version breaks tests.

File: prestodb/client.py (hunk 2 above)
Comment (dcfocus) [reply prestodb/client.py]: Updated.

File: .travis.yml (hunk 1 above)
Comment (ChunxuTang) [reply .travis.yml:7]: I see. Thanks.

**Comment by ChunxuTang** [comment]
lgtm. Thanks for your work!

**Comment by dcfocus** [comment]
Duplicated to #97
